src/game/Tactical/PathAI.cc

```diff
@@ -2533,62 +2533,77 @@ UINT8 InternalDoorTravelCost(const SOLDIERTYPE* pSoldier, INT32 iGridNo, UINT8 u
 			case TRAVELCOST_DOOR_OPEN_HERE:
 				fDoorIsObstacleIfClosed = FALSE;
 				iDoorGridNo = iGridNo;
+				ubReplacementCost = TRAVELCOST_DOOR_OPEN_HERE;
 				break;
 			case TRAVELCOST_DOOR_OPEN_N:
 				fDoorIsObstacleIfClosed = FALSE;
 				iDoorGridNo = iGridNo + DirIncrementer[ NORTH ];
+				ubReplacementCost = TRAVELCOST_DOOR_OPEN_N;
 				break;
 			case TRAVELCOST_DOOR_OPEN_NE:
 				fDoorIsObstacleIfClosed = FALSE;
 				iDoorGridNo = iGridNo + DirIncrementer[ NORTHEAST ];
+				ubReplacementCost = TRAVELCOST_DOOR_OPEN_NE;
 				break;
 			case TRAVELCOST_DOOR_OPEN_E:
 				fDoorIsObstacleIfClosed = FALSE;
 				iDoorGridNo = iGridNo + DirIncrementer[ EAST ];
+				ubReplacementCost = TRAVELCOST_DOOR_OPEN_E;
 				break;
 			case TRAVELCOST_DOOR_OPEN_SE:
 				fDoorIsObstacleIfClosed = FALSE;
 				iDoorGridNo = iGridNo + DirIncrementer[ SOUTHEAST ];
+				ubReplacementCost = TRAVELCOST_DOOR_OPEN_SE;
 				break;
 			case TRAVELCOST_DOOR_OPEN_S:
 				fDoorIsObstacleIfClosed = FALSE;
 				iDoorGridNo = iGridNo + DirIncrementer[ SOUTH ];
+				ubReplacementCost = TRAVELCOST_DOOR_OPEN_S;
 				break;
 			case TRAVELCOST_DOOR_OPEN_SW:
 				fDoorIsObstacleIfClosed = FALSE;
 				iDoorGridNo = iGridNo + DirIncrementer[ SOUTHWEST ];
+				ubReplacementCost = TRAVELCOST_DOOR_OPEN_SW;
 				break;
 			case TRAVELCOST_DOOR_OPEN_W:
 				fDoorIsObstacleIfClosed = FALSE;
 				iDoorGridNo = iGridNo + DirIncrementer[ WEST ];
+				ubReplacementCost = TRAVELCOST_DOOR_OPEN_W;
 				break;
 			case TRAVELCOST_DOOR_OPEN_NW:
 				fDoorIsObstacleIfClosed = FALSE;
 				iDoorGridNo = iGridNo + DirIncrementer[ NORTHWEST ];
+				ubReplacementCost = TRAVELCOST_DOOR_OPEN_NW;
 				break;
 			case TRAVELCOST_DOOR_OPEN_N_N:
 				fDoorIsObstacleIfClosed = FALSE;
 				iDoorGridNo = iGridNo + DirIncrementer[ NORTH ] + DirIncrementer[ NORTH ];
+				ubReplacementCost = TRAVELCOST_DOOR_OPEN_N_N;
 				break;
 			case TRAVELCOST_DOOR_OPEN_NW_N:
 				fDoorIsObstacleIfClosed = FALSE;
 				iDoorGridNo = iGridNo + DirIncrementer[ NORTHWEST ] + DirIncrementer[ NORTH ];
+				ubReplacementCost = TRAVELCOST_DOOR_OPEN_NW_N;
 				break;
 			case TRAVELCOST_DOOR_OPEN_NE_N:
 				fDoorIsObstacleIfClosed = FALSE;
 				iDoorGridNo = iGridNo + DirIncrementer[ NORTHEAST ] + DirIncrementer[ NORTH ];
+				ubReplacementCost = TRAVELCOST_DOOR_OPEN_NE_N;
 				break;
 			case TRAVELCOST_DOOR_OPEN_W_W:
 				fDoorIsObstacleIfClosed = FALSE;
 				iDoorGridNo = iGridNo + DirIncrementer[ WEST ] + DirIncrementer[ WEST ];
+				ubReplacementCost = TRAVELCOST_DOOR_OPEN_W_W;
 				break;
 			case TRAVELCOST_DOOR_OPEN_SW_W:
 				fDoorIsObstacleIfClosed = FALSE;
 				iDoorGridNo = iGridNo + DirIncrementer[ SOUTHWEST ] + DirIncrementer[ WEST ];
+				ubReplacementCost = TRAVELCOST_DOOR_OPEN_SW_W;
 				break;
 			case TRAVELCOST_DOOR_OPEN_NW_W:
 				fDoorIsObstacleIfClosed = FALSE;
 				iDoorGridNo = iGridNo + DirIncrementer[ NORTHWEST ] + DirIncrementer[ WEST ];
+				ubReplacementCost = TRAVELCOST_DOOR_OPEN_NW_W;
```

Comment:
why not just set `ubReplacementCost = ubMovementCost;` before the switch?

Reply:
Before the switch case there is 
`ubReplacementCost = TRAVELCOST_OBSTACLE;`

Why would you want to override it with the movementcost?

Reply:
So you don't need to reassign it for each case in the switch, just the special ones from before and TRAVELCOST_OBSTACLE for the default case. All the new additions are like:
```
case A:
  ...
  ubReplacementCost = A
```
which is not as sleek as it could be.